t/14-grutasks-git.t

```diff
@@ -266,11 +272,36 @@ subtest 'git_update_all' => sub {
         $testdir->child("$path/.git")->make_path;
     }
     local $ENV{OPENQA_BASEDIR} = $workdir;
+    local $ENV{OPENQA_GIT_CLONE_RETRIES} = 4;
+    local $ENV{OPENQA_GIT_CLONE_RETRIES_BEST_EFFORT} = 2;
     my $minion = $t->app->minion;
     my $result = $t->app->gru->enqueue_git_update_all;
     my $job = $minion->job($result->{minion_id});
     my $args = $job->info->{args}->[0];
+    my $gru_id = $result->{gru_id};
+    my $gru_task = $gru_tasks->find($gru_id);
     is_deeply [sort keys %$args], \@clones, 'job args as expected';
+    isnt $gru_task, undef, 'gru task created';
+
+    subtest 'error handling and retry behavior' => sub {
+        # assume an openQA job is blocked on the Gru task
+        my $blocked_job = $jobs->create({TEST => 'blocked-job'});
+        $gru_task->jobs->create({job_id => $blocked_job->id});
+
+        # perform job, it'll fail via the mocked _git_clone function
+        $minion->foreground($job->id);    # already counts as retry
+        is $job->info->{state}, 'inactive', 'job failed but set to inactive to be retried';
+        is $gru_task->jobs->count, 1, 'openQA job not unblocked after first try';
+
+        # retry the job now
+        $minion->foreground($job->id);
+        is $job->info->{state}, 'inactive', 'job failed but again set to inactive to be retried';
+        is $gru_task->jobs->count, 0, 'openQA job unblocked after second try';
+
+        combined_like { $minion->foreground($job->id) } qr/fake disconnect/, 'error logged';
+        is $job->info->{state}, 'failed', 'job failed for real after retries exhausted';
+      }
+      if $gru_task;
```

Comment:
Looks like the unit test never sets `git_auto_update_method` so `strict` is never tested?

Reply:
That's true. Just like we never test for "no" in "yes"/"no" settings. Everything that is not "yes" is simply "no". And here it is similar, so everything that is not "best-effort" is "strict". I could of course add some kind of validation in the config parser and print a warning for invalid values. (That's not how we usually handle this, though.)

Reply:
Ah. I wasn't thinking about validation. Just to cover the case where it is set to `strict` and doesn't use the new logic.